app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/CurrentWorkoutActivity.java

```diff
@@ -65,10 +67,46 @@
     // Text fields in the activity
     private TextView mCounter;
 
-    // Object for scheduling fake laps counted.
-//    private Handler mHandler = new Handler();
-
     private TextView mDebugConnectLabel;
+    private int mLapCount;
+
+    private void pause() {
+        // If we're already paused, do nothing.
+        if (isPaused)
+            return;
+
+        timeSwapBuff += timeInMilliseconds;
+        customHandler.removeCallbacks(updateTimerThread);
+
+        startResumeButton.setEnabled(true);
+        pauseButton.setEnabled(false);
+
+        isPaused = true;
+        mBleService.stopRssiRequests();
+    }
+
+    private void saveWorkout() {
+        Workout workout = new Workout();
+        workout.setDeviceMAC(loadDeviceAddress());
+        workout.setStartDate(startTime);
+        workout.setEndDate(new Date());
```

Comment:
Might be good to set end date as the time we last paused, if the workout is currently puased.